WordPress/gutenberg
# Add Block Deprecation Warning automation task

Related: https://github.com/WordPress/gutenberg/projects/24#card-36342527

This pull request seeks to add a new project management automation task, which serves to add a comment to a pull request which is opened and contains revisions to a core block implementation's `save.js` file, but does not include an accompanying `deprecated.js` revision.

Currently, the comment is written with the following message:

>It appears that this pull request is modifying the generated saved markup of a core block, but does not include an accompanying deprecation. It is typically expected that a deprecation should be implemented, as otherwise it could cause existing content to become invalid.
>
>If you've already taken remediation steps or a deprecation should not apply, you can safely ignore this message.
>
>[See this  automation's documentation](https://github.com/WordPress/gutenberg/blob/master/packages/project-management-automation/lib/tasks/block-deprecation-warning/README.md) for more information, including references to related resources.

See also the included task README.md file, which documents the task, rationale, and related resources.

**Testing Instructions:**

Ensure unit tests pass:

```
npm run test-unit packages/project-management-automation/lib/tasks/block-deprecation-warning/test/index.js
```

Since this automation runs at pull request opening, it will be difficult to verify real-world usage until the point of merge. That said, it may be run on _this_ pull request (unclear).
